idb/postgres/internal/encoding/types.go

```diff
@@ -0,0 +1,24 @@
+package encoding
+
+import (
+	sdk_types "github.com/algorand/go-algorand-sdk/types"
+
+	"github.com/algorand/indexer/types"
+)
+
+type assetParams struct {
```

Comment:
I don't like the split of types.go/encode.go/decode.go
I'd rather see one type declaration and its encode and decode in one file (OOP style). Easier to make sure encode and decode match that way. Easier to find the type being referred to.

Reply:
This split makes a lot more sense in the context of Tolik's rewrite, this pattern is used to wrap many more objects than I've done here.

I would probably opt for a single file, only because it would let you use simple text searches to find everything. But don't have a strong opinion either way. For now, I'd like to match his PR to make the merge slightly easier, maybe we can postpone this discussion for the rewrite PR?

Reference: https://github.com/algorand/indexer/blob/tolik/acct-rewrite4/idb/postgres/internal/encoding/types.go

Reply:
I won't block over it but I still don't like it. I still want type-encode-decode bundled together in one file. `types.go` makes the most sense when the only thing happening to them is via reflection (e.g. go-codec loading and storing serialization forms).